src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1373,53 +1368,25 @@ public void unpublishFeedbackSession(FeedbackSessionAttributes sessionToUnpublis
     }
 
     /**
-     * Deletes the feedback sessions in the course specified. The delete
-     * is cascaded, and feedback questions, feedback responses, and
-     * feedback response comments in the course are deleted.
-     */
-    public void deleteFeedbackSessionsForCourseCascade(String courseId) {
-        frcLogic.deleteFeedbackResponseCommentsForCourse(courseId);
-        frLogic.deleteFeedbackResponsesForCourse(courseId);
-        fqLogic.deleteFeedbackQuestionsForCourse(courseId);
-        deleteFeedbackSessionsForCourse(courseId);
-    }
-
-    /**
-     * Deletes all feedback sessions the course specified. This is
-     * a non-cascade delete.
-     *
-     * <p>The responses, questions and the comments of the responses
-     * should be handled.
-     */
-    public void deleteFeedbackSessionsForCourse(String courseId) {
-        fsDb.deleteFeedbackSessionsForCourse(courseId);
-    }
-
-    /**
-     * Permanently deletes a specific feedback session in Recycle Bin, and all its questions and responses.
+     * Deletes a feedback session cascade to its associated questions, responses and comments.
      */
     public void deleteFeedbackSessionCascade(String feedbackSessionName, String courseId) {
+        AttributesDeletionQuery query = AttributesDeletionQuery.builder()
+                .withCourseId(courseId)
+                .withFeedbackSessionName(feedbackSessionName)
+                .build();
+        fqLogic.deleteFeedbackQuestions(query);
+        frLogic.deleteFeedbackResponses(query);
+        frcLogic.deleteFeedbackResponseComments(query);
```

Comment:
Similarly here; response comments should be deleted before the responses, which should be deleted before the questions?